app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java

```diff
@@ -654,12 +667,28 @@ public String getFragmentTag() {
         return FRAGMENT_TAG;
     }
 
+    public boolean checkPermissions() {
+
+        if (_document != null) {
+            final File file = _document.getFile();
+
+            if (_shareUtil.isUnderStorageAccessFolder(file) && _shareUtil.getStorageAccessFrameworkTreeUri() == null) {
+                _shareUtil.showMountSdDialog(getActivity());
+            }
+
+            final boolean permok =  _document.testCreateParent() && _shareUtil.canWriteFile(file);
+            _textSdWarning.setVisibility(permok ? View.GONE : View.VISIBLE);
+            return permok;
+        }
+        return false;
```

Comment:
In this past (and in this PR too) we are checking permissions all the time instead of just once on the initial load. I guess the reason is that the filesytem could change while the document is open. Do you think it makes sense to keep this? Or is it sufficient to just display a warning to the user on  start and then not check?